job-server/src/spark.jobserver/AkkaClusterSupervisorActor.scala

```diff
@@ -205,10 +205,15 @@ class AkkaClusterSupervisorActor(daoActor: ActorRef) extends InstrumentedActor {
           return
       }
 
+    var driverMemory = "0"
+    if (contextConfig.hasPath("spark.driver") && contextConfig.hasPath("spark.driver.memory")) {
```

Comment:
I think you can eliminate the check entirely by doing something like

    Try(config.getString("spark.driver.memory")).getOrElse(....)

or use match etc.   Lately I've been using another library called Ficus which can return Options.  Either way is much DRYer than having to check beforehand.